irr/src/os.h

```diff
@@ -10,6 +10,21 @@
 #include "ILogger.h"
 #include "ITimer.h"
 
+// CODE_UNREACHABLE(): Invokes undefined behavior for unreachable code optimization
+#if defined(__cpp_lib_unreachable)
+#define CODE_UNREACHABLE() std::unreachable()
```

Comment:
missing `<utility>` include

Reply:
Done in 943ea3e57

BTW, would we need a C++23 unit test?